api/src/main/java/io/grpc/ServerCall.java

```diff
@@ -209,6 +209,10 @@ public void setCompression(String compressor) {
     // noop
   }
 
+  public void setOnReadyThreshold(int numBytes) {
```

Comment:
We're going to want some javadoc here. We probably also need to say that the value must be positive.

Let's also document that it should only be used before sending any messages. Changing it in the middle of the RPC doesn't seem all that necessary and your current implementation doesn't handle changing it after messages are sent (it doesn't handle isReady false → true transition by notifying onReady when increasing the threshold). That would also _almost_ allow removing the GuardedBy on `onReadyThreshold`. (We'd still keep it GuardedBy because isReady() is thread-safe and there'd be a benign data race. The lock will make TSAN happy.)

Reply:
I agree that this should be set before any messages get sent.